pages/home.js

```diff
@@ -33,6 +36,31 @@ export default function Home(props) {
   const handleJoinSubmit = (e) => {
     e.preventDefault()
   }
+
+  let onCreateHandler = (e) => {
+    //prevent default behaviour of form
+    e.preventDefault()
+
+    //Check if name is empty
+    if (owner.value.trim() === '') {
+      setValid(false)
+    }
+    //Check if name contains special characters
+    else if (!/^[a-zA-Z0-9]+$/.test(owner.value)) {
+      setValid(false)
+    } else {
+      setValid(true)
+    }
+
+    //If name is valid, create new room
+    if (isValid) {
+      addRoom({ variables: { name: e.target.owner.value } }).then((res) =>
+        router.push({
+          pathname: `/room/${res.data.addRoom.id}`,
+        })
+      )
+    }
+  }
```

Comment:
Something about this logic isn't working and I'm not sure why.
setValid(false) is setting the state to false but isValid returns true.

Another thing to consider is if the addRoom request fails we can now catch it and display the ErrorLabel component. addRoom was returning a 400 error for bad request (probably some server work that was done) and hanging